src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/http/MeshClient.java

```diff
@@ -86,21 +103,56 @@ public void acknowledgeMessage(String messageId) {
 
     @SneakyThrows
     public List<String> getInboxMessageIds() {
+        final var loggingName = "Check inbox";
         try (CloseableHttpClient client = new MeshHttpClientBuilder(meshConfig).build()) {
-            try (CloseableHttpResponse response = client.execute(meshRequests.getMessageIds())) {
+            var request = meshRequests.getMessageIds();
+            logRequest(loggingName, request);
+            try (CloseableHttpResponse response = client.execute(request)) {
+                logResponse(loggingName, response);
                 if (response.getStatusLine().getStatusCode() != HttpStatus.OK.value()) {
                     throw new MeshApiConnectionException("Couldn't receive MESH message list",
-                            HttpStatus.OK,
-                            HttpStatus.valueOf(response.getStatusLine().getStatusCode()));
+                        HttpStatus.OK,
+                        HttpStatus.valueOf(response.getStatusLine().getStatusCode()));
                 }
-                return Arrays.asList(parseInto(MeshMessages.class, response).getMessageIDs());
+                var meshMessages = parseInto(MeshMessages.class, response, loggingName);
+                return Arrays.asList(meshMessages.getMessageIDs());
             }
         }
     }
 
-    private <T> T parseInto(Class<T> clazz, CloseableHttpResponse response) throws IOException {
+    private <T> T parseInto(Class<T> clazz, CloseableHttpResponse response, String loggingName) throws IOException {
         ObjectMapper objectMapper = new ObjectMapper();
-        JsonParser parser = objectMapper.reader().createParser(EntityUtils.toString(response.getEntity()));
-        return objectMapper.readValue(parser, clazz);
+        var content = EntityUtils.toString(response.getEntity());
+        LOGGER.debug("MESH '{}' response content: {}", loggingName, content);
+        return objectMapper.readValue(content, clazz);
+    }
+
+    @SneakyThrows
+    private void logRequest(String type, HttpRequest request) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("MESH '{}' request line: {}", type, request.getRequestLine());
+            LOGGER.debug("MESH '{}' request headers: {}", type, request.getAllHeaders());
+
+            if (request instanceof HttpEntityEnclosingRequest) {
+                var entity = ((HttpEntityEnclosingRequest) request).getEntity();
+                if (entity != null) {
+                    LOGGER.debug("MESH '{}' request content line: {}", type, entity);
```

Comment:
Not logging content?
If you would log content, remember that stream needs rewinding after it's read so further production code can use it

Reply:
comment on the next line explains this